### PR TITLE
[#629] Password Reset Issue Fixed (connects #629)

### PR DESCRIPTION
**Pull request by nirojdyola**

### CODE REVIEW

File: app/MyApp/app/Router.js [hunk 1]
```diff
@@ -4738,35 +4738,17 @@ $(function() {
...
-                    async: false
-                })
-                var mailview = new App.Views.MailView({
-                    collection: mymail,
-                    community_code: code,
-                    nationName: Bellname
-                })
-            }
-			
+			mymail.fetch({
```
Comment (dogi) [inline app/MyApp/app/Router.js]: @nirojdyola looks like this are tabs ... please fix this whitespace issue ;)

File: app/MyApp/app/models/Configuration.js [hunk 2]
```diff
@@ -101,7 +101,11 @@ $(function() {
                 type: 'Text' // This attribute will be removed when all the communities will have 'easy-install' work.
             },
             countDoubleUpdate: 'Number', //Saves status of community's update.
-            kind: 'Text' // Saves kind of document according to corresponding db's.Mostly used in couch db views.
+            kind: 'Text', // Saves kind of document according to corresponding db's.Mostly used in couch db views.
```
Comment (dogi) [inline app/MyApp/app/models/Configuration.js:104]: same whitespace issue here

File: app/MyApp/app/views/CommunityConfigurationsForm.js [hunk 3]
```diff
@@ -23,6 +23,7 @@ $(function() {
             }
             vars["nations"] = [];
             vars["languages"] = [];
+			vars["passwordResetDuration"] = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'];
```
Comment (dogi) [inline app/MyApp/app/views/CommunityConfigurationsForm.js]: and here

File: app/MyApp/app/views/MailView.js [hunk 4]
```diff
@@ -406,38 +394,38 @@ $(function() {
...
-                        mailView = this;
-                        if(App.languageDict.get('directionOfLang').toLowerCase()==="right"){
-                            $('#viewDelCol').find('td').eq(2).attr("align","left")
-                        }
-                        else {
-                            $('#viewDelCol').find('td').eq(2).attr("align","right")
-                        }
-                    }
-                }else{
+				if(vars.type == "PasswordReset"){
```
Comment (dogi) [inline app/MyApp/app/views/MailView.js]: here

File: app/MyApp/app/views/CommunityConfigurationsForm.js [hunk 5]
```diff
@@ -289,7 +291,8 @@ $(function() {
                 kind: 'Community',
                 //Temporarily adding these attributes
                 Name: $.trim($('#community-name').val()),
-                Code: $.trim($('#community-code').val())
+                Code: $.trim($('#community-code').val()),
```
Comment (dogi) [inline app/MyApp/app/views/CommunityConfigurationsForm.js:294]: here

File: app/MyApp/app/views/CommunityConfigurationsForm.js [hunk 6]
```diff
@@ -95,6 +96,7 @@ $(function() {
             $('#leader-lastname').val(that.model.get('superManagerLastName'));
             $('#leader-phone').val(that.model.get('superManagerPhone'));
             $('#leader-email').val(that.model.get('superManagerEmail'));
+			$('#pwd_reset_duration').val(that.model.get('passwordResetDuration'));
```
Comment (dogi) [inline app/MyApp/app/views/CommunityConfigurationsForm.js]: here

File: app/MyApp/app/views/MailView.js [hunk 7]
```diff
@@ -406,38 +394,38 @@ $(function() {
...
                     $("#" + viewId).click(this.viewButton)
-                    mailView = this;
-                    if(App.languageDict.get('directionOfLang').toLowerCase()==="right"){
-                        $('#viewDelCol').find('td').eq(2).attr("align","left")
-                    }
-                    else {
-                        $('#viewDelCol').find('td').eq(2).attr("align","right")
-                    }
-                }
+				}
```
Comment (dogi) [inline app/MyApp/app/views/MailView.js]: here

File: app/MyApp/app/views/MailView.js [hunk 8]
```diff
@@ -474,18 +462,10 @@ $(function() {
...
-                    status: obj.status,
-                    skip: 0
-                });
-            }else{
-                var newCollection = new App.Collections.Mails({
-                    receiverId: $.cookie('Member._id'),
-                    unread: obj.unopen
-                })
-            }
+			var newCollection = new App.Collections.Mails({
```
Comment (dogi) [inline app/MyApp/app/views/MailView.js]: here

File: app/MyApp/app/views/MailView.js [hunk 9]
```diff
@@ -732,35 +712,87 @@ $(function() {
         },
 		
         acceptButton: function(e){
-            var that = e.currentTarget;
-            var modelNo = $(that).closest('tr').index();
-            var acceptedBy = $.cookie('Member.login');
+			var modelNo = e.currentTarget.value;
```
Comment (dogi) [inline app/MyApp/app/views/MailView.js]: here

File: app/MyApp/app/views/MailView.js [hunk 10]
```diff
@@ -732,35 +712,87 @@ $(function() {
...
-            var login_member = $(that).parent().parent().find('td#.member').text();
-            var subject =  '<td>' + App.languageDict.attributes.Email_PasswordReset_Subject + ' | ' + login_member + '</td><td align="center">' + login_member + '</td><td id="viewDelCol">'+ App.languageDict.attributes.Msg_Email_ApprovedBy +' '+ acceptedBy +'</td></tr>';
-            model.set('subject',subject);
-            model.set('status',1);
-            model.set('member_status',0);
-            if(model.save()){
-                alert(App.languageDict.attributes.Msg_PasswordReset_Email_Approved);
-                $(that).closest('tr').remove();
-            }
+			var senderId = model.get("senderId");
```
Comment (dogi) [inline app/MyApp/app/views/MailView.js]: here

File: app/MyApp/app/views/MailView.js [hunk 11]
```diff
@@ -732,35 +712,87 @@ $(function() {
...
-            var login_member = $(that).parent().parent().find('td#.member').text();
-            var subject =  '<td>' + App.languageDict.attributes.Email_PasswordReset_Subject + ' | ' + login_member + '</td><td align="center">' + login_member + '</td><td id="viewDelCol">' + App.languageDict.attributes.Msg_Email_RejectedBy + ' '+ rejectedBy +'</td></tr>';
-            model.set('subject',subject);
-            model.set('status',2);
-            model.set('member_status',0);
-            if(model.save()){
-                alert(App.languageDict.attributes.Msg_PasswordReset_Email_Rejected);
-                $(that).closest('tr').remove();
-            }
+			var senderId = model.get("senderId");
```
Comment (dogi) [inline app/MyApp/app/views/MailView.js]: here

File: init_docs/Text.json [hunk 12]
```diff
@@ -917,5 +915,8 @@
     "PasswordResetLogin": "Password Reset and logged in",
     "Msg_Re_ResetPassword": "You have recently updated password. Do you want to reset your password again ?",
     "Msg_Success_ResetPassword_ChangePassword": "Your request has been approved. You can change your password.",
-    "Msg_Confirm_ResetPassword_ChangePassword": "You have recently updated password. Do you want to reset your password again ?If you click No then it will resend the password email again."
+    "Msg_Confirm_ResetPassword_ChangePassword": "You have recently updated password. Do you want to reset your password again ?If you click No then it will resend the password email again.",
```
Comment (dogi) [inline init_docs/Text.json]: and here

File: app/MyApp/index.html [hunk 13]
```diff
@@ -787,6 +786,19 @@
                         <input style='width:214px;'  type="email" Required id="leader-email" name="leader-email" /><br />
                     </div>
                 </div>
+				<div>
```
Comment (dogi) [inline app/MyApp/index.html]: here

File: app/MyApp/app/views/MemberLoginForm.js [hunk 14]
```diff
@@ -366,101 +370,163 @@ $(function() {
                 });
         },
 		
-		updatePasswordResetEmail: function(member){
+		updateLoggedInPasswordResetEmail: function(member){
```
Comment (dogi) [inline app/MyApp/app/views/MemberLoginForm.js]: here

File: app/MyApp/app/views/MemberLoginForm.js [hunk 15]
```diff
@@ -366,101 +370,163 @@ $(function() {
...
 			if (member.get('status') == "active") {
 				 var login_ID = member.get('_id');
 				 var login_member = member.get('login');
 				 var mailCollections = new App.Collections.Mails({
 					senderId : login_ID,
-					type : "PasswordReset",
-					skip: 0
+					type : "PasswordReset"
```
Comment (dogi) [inline app/MyApp/app/views/MemberLoginForm.js]: here

File: app/MyApp/app/views/MemberLoginForm.js [hunk 16]
```diff
@@ -366,101 +370,163 @@ $(function() {
...
-								async: false
-							})
 							var currentdate = new Date();
-							if(mail_status == 0 && member_status == 0){
-								var subject =  '<td>' + 'Reset automatically' + ' | ' + login_member + '</td><td align="center" class="member">' + login_member + '</td><td id="viewDelCol">'+ App.languageDict.attributes.Remembered_AND_Login +'</td></tr>';
-							}else if(mail_status == 1 && member_status == 0){
-								var subject =  '<td>' + 'Reset automatically' + ' | ' + login_member + '</td><td align="center" class="member">' + login_member + '</td><td id="viewDelCol">'+ App.languageDict.attributes.Remembered_AND_Login +'</td></tr>';
-							}else if(mail_status == 1 && member_status == 1){
-								var subject =  '<td>' + 'Reset automatically' + ' | ' + login_member + '</td><td align="center" class="member">' + login_member + '</td><td id="viewDelCol">'+ App.languageDict.attributes.PasswordResetLogin +'</td></tr>';
+							var subject;
```
Comment (dogi) [inline app/MyApp/app/views/MemberLoginForm.js]: here

File: app/MyApp/app/views/MemberLoginForm.js [hunk 17]
```diff
@@ -366,101 +370,163 @@ $(function() {
...
 						}
 					}
 				});
 			 }
 		},
 		
         checkPasswordResetEmail: function(member){
-            var flag = 1 ;
+            var flag = "" ;
+			var self = this;
```
Comment (dogi) [inline app/MyApp/app/views/MemberLoginForm.js]: here

File: app/MyApp/app/views/MemberLoginForm.js [hunk 18]
```diff
@@ -366,101 +370,163 @@ $(function() {
...
+                    type : "PasswordReset"
                 });
                 mailCollections.fetch({
                     async : false,
                     success: function(){
                         var mail_id;
                         var mail_status;
                         var member_status;
-                        console.log(mailCollections);
+						var mail_dateTime;
```
Comment (dogi) [inline app/MyApp/app/views/MemberLoginForm.js]: here

File: app/MyApp/app/views/MemberLoginForm.js [hunk 19]
```diff
@@ -366,101 +370,163 @@ $(function() {
...
+						if(passwordResetDuration == undefined)
+							duration = 0;
+						else
+							duration = passwordResetDuration;
                         if(mail_status == 0){
-                            //still Pending
-                            alert(App.languageDict.attributes.Msg_Pending_PasswordReset);
-                            return false;
-                        }
+							if(hr > duration && duration > 0){
```
Comment (dogi) [inline app/MyApp/app/views/MemberLoginForm.js]: here

File: app/MyApp/app/views/MemberLoginForm.js [hunk 20]
```diff
@@ -366,101 +370,163 @@ $(function() {
...
+							}
+                        }else if(mail_status == 2){
+							//Rejected
+							flag = "Rejected";
+						}
                         if(mail_status == 1 && member_status == 1){
                             var check_confirm = confirm(App.languageDict.attributes.Msg_Re_ResetPassword);
                             if(check_confirm)
-                            flag = 2;
+								flag = "PasswordReset";
```
Comment (dogi) [inline app/MyApp/app/views/MemberLoginForm.js]: here

File: app/MyApp/app/views/MemberLoginForm.js [hunk 21]
```diff
@@ -366,101 +370,163 @@ $(function() {
...
                             else
-                            flag = 0;
+								flag = "Nothing";
```
Comment (dogi) [inline app/MyApp/app/views/MemberLoginForm.js]: here

File: app/MyApp/app/views/MemberLoginForm.js [hunk 22]
```diff
@@ -366,101 +370,163 @@ $(function() {
...
-                                flag = 0;
+                                flag = "Nothing";
                         }
                     }
                 });
             }
             return flag;
         },
 		
+		updatePasswordResetEmail: function(mailCollections){
```
Comment (dogi) [inline app/MyApp/app/views/MemberLoginForm.js]: here

File: app/MyApp/app/views/MemberLoginForm.js [hunk 23]
```diff
@@ -366,101 +370,163 @@ $(function() {
...
+						flag = "Msg_AutomaticApproved_PasswordReset";
+					}
+				}
+			}
+			return flag;
+		},
+		
         sendPasswordResetEmail: function(member){
             //check the login input
+			var self = this;
```
Comment (dogi) [inline app/MyApp/app/views/MemberLoginForm.js]: here

File: app/MyApp/app/views/MemberLoginForm.js [hunk 24]
```diff
@@ -480,110 +546,119 @@ $(function() {
...
-                                                        mail.set("type", "PasswordReset");
-                                                        mail.set("sentDate", currentdate);
-                                                        if(mail.save()){
-                                                            alert(App.languageDict.attributes.Msg_Request_PasswordReset);
-                                                        }
-                                                    }
-                                                }
-                                            });
-                                        }
+										var mail = new App.Models.Mail();
```
Comment (dogi) [inline app/MyApp/app/views/MemberLoginForm.js]: here

**Review by dogi — CHANGES_REQUESTED** [review]
@nirojdyola think you should configure your editor so that tabs get converted to spaces ;)

File: app/MyApp/app/Home.css [hunk 25]
```diff
@@ -159,45 +159,45 @@ background-color:#E1E1E1;
...
-	font-weight: 900;
+     font-size: 30px;
+     padding-top: 12px;
+    font-weight: 900;
 }
 .stu-nav-table  
 {
-padding:5 5 0 5;	
-width:100%;	
+padding:5 5 0 5;
```
Comment (dogi) [inline app/MyApp/app/Home.css:171]: oops trailing spaces ;)

File: app/MyApp/app/Home.css [hunk 26]
```diff
@@ -159,45 +159,45 @@ background-color:#E1E1E1;
...
+padding:5 5 0 5;    
+width:100%;    
 height:280px;
 padding-top:5px;
 }
 .page-bottom-navtable 
 {
 width: 943px;
-height:50px;	
+height:50px;
```
Comment (dogi) [inline app/MyApp/app/Home.css:179]: and here

File: app/MyApp/app/Home.css [hunk 27]
```diff
@@ -159,45 +159,45 @@ background-color:#E1E1E1;
...
-height:50px;	
+height:50px;    
 margin-bottom: 50px;
 }
 
 .stu-nav-title > a{
-	color: #FFA500;
-	
+    color: #FFA500;
+
```
Comment (dogi) [inline app/MyApp/app/Home.css:185]: here

File: app/MyApp/app/Home.css [hunk 28]
```diff
@@ -159,45 +159,45 @@ background-color:#E1E1E1;
...
 .stu-nav-title
 {
 border: 1.5px solid darkgrey;
-	font-weight:bolder;
-	text-align:left;	
-	font-family: verdana;
-	font-size: large;
-	padding: 8px 8px;
+    font-weight:bolder;
+    text-align:left;
```
Comment (dogi) [inline app/MyApp/app/Home.css:192]: here

File: app/MyApp/app/Home.css [hunk 29]
```diff
@@ -453,27 +453,27 @@ margin-left:40px;
...
 }
 
 .navbar-inverse .nav > li > a
 {
-	border: none;
-	font-size: 15px;
-	padding: 16px 20px 15px 0px;	
+    border: none;
+    font-size: 15px;
+    padding: 16px 20px 15px 0px;
```
Comment (dogi) [inline app/MyApp/app/Home.css:476]: here

File: app/MyApp/app/Home.css [hunk 30]
```diff
@@ -487,36 +487,36 @@ padding-top: 5px;
...
-	}
+    font-weight: bolder;
+        height: 40px;
+    font-size: 30px;
+    }
 .news-table tr
 {
- border-top: 2px dashed #000000;	
-	font-size: 17px;
+ border-top: 2px dashed #000000;
```
Comment (dogi) [inline app/MyApp/app/Home.css:502]: here

File: app/MyApp/app/Home.css [hunk 31]
```diff
@@ -487,36 +487,36 @@ padding-top: 5px;
...
 .news-table tr
 {
- border-top: 2px dashed #000000;	
-	font-size: 17px;
+ border-top: 2px dashed #000000;    
+    font-size: 17px;
   font-weight: 900;
   height: 47px;
-  overflow: hidden;	
+  overflow: hidden;
```
Comment (dogi) [inline app/MyApp/app/Home.css:506]: here

File: app/MyApp/app/Home.css [hunk 32]
```diff
@@ -529,15 +529,15 @@ padding-top: 5px;
...
 }
 
 .search-bottom-nav
 {
-	margin-left: 5%;
+    margin-left: 5%;
     overflow: hidden;
     width: 90%;
-	margin-right: 5%;	
+    margin-right: 5%;
```
Comment (dogi) [inline app/MyApp/app/Home.css:540]: here

File: app/MyApp/app/Router.js [hunk 33]
```diff
@@ -305,7 +305,7 @@ $(function() {
                     $('#syncStatus').closest('div').show();
                 },
                 error: function (status) {
-                	$('#syncStatus').closest('div').hide();
+                    $('#syncStatus').closest('div').hide();
```
Comment (dogi) [inline app/MyApp/app/Router.js:308]: here

Comment (dogi) [reply app/MyApp/app/Router.js:308]: oops my error ;) not here I mean

**Review by lmmrssa — APPROVED** [review]

File: app/MyApp/app/Router.js [hunk 34]
```diff
@@ -6669,71 +6651,71 @@ $(function() {
...
+                                        }
+                                    })
+                                },
+                                error: function(err) {
+                                    console.log(err);
+                                }
+                            })
+                            return true;
+                        }
+
```
Comment (dogi) [inline app/MyApp/app/Router.js:6709]: here

File: app/MyApp/app/Router.js [hunk 35]
```diff
@@ -6669,71 +6651,71 @@ $(function() {
...
+                    });
                 },
                 error: function(err) {
                     console.log(err);
                 }
             })
             applyCorrectStylingSheet(App.languageDict.get('directionOfLang'));
         },
-		
+
```
Comment (dogi) [inline app/MyApp/app/Router.js:6718]: here

**Review by dogi — CHANGES_REQUESTED** [review]
sorry there are also trailing whitespaces

**Review by Rupesh87 — APPROVED** [review]

**Pull request merged**